src/main.rs

```diff
@@ -39,19 +42,26 @@ fn run(args: &ArgMatches) -> Result<(), Error> {
         &simple_format
     };
 
+    let scale = value_t!(args.value_of("scale"), u32).unwrap_or_else(|e| match e.kind {
+        ErrorKind::ArgumentNotFound => DEFAULT_SCALE,
+        _ => error(&format!("{}", e)),
+    });
+    let preview_size =
+        value_t!(args.value_of("preview_size"), u32).unwrap_or_else(|e| match e.kind {
+            ErrorKind::ArgumentNotFound => DEFAULT_PREVIEW_SIZE,
+            _ => error(&format!("{}", e)),
+        });
+
     let selection = args.values_of("selection").and_then(|mut v| {
         v.next()
             .map_or(Some(Selection::Clipboard), |v| v.parse::<Selection>().ok())
     });
     let use_selection = selection.is_some();
     let background = std::env::var("XCOLOR_FOREGROUND").is_err();
 
-    let use_preview = !args.is_present("no_preview");
-    let use_shape = std::env::var("XCOLOR_DISABLE_SHAPE").is_err();
-
     let mut in_parent = true;
 
-    let (conn, screen) = Connection::connect(None)?;
+    let (conn, screen) = Connection::connect_with_xlib_display()?;
```

Comment:
We need to connect with the `xlib_display` because we're using the `Xcursor` APIs and they require a pointer to a Display. Connecting the normal way connected without retrieving the pointer.